static/panes/pane.ts

```diff
@@ -231,39 +218,55 @@ export abstract class Pane<E extends monaco.editor.IEditor, S> {
         }
     }
 
-    protected onDidChangeCursorSelection(event: monaco.editor.ICursorSelectionChangedEvent) {
-        if (this.isAwaitingInitialResults) {
-            this.selection = event.selection;
-            this.updateState();
-        }
-    }
-
     protected onSettingsChange(settings: SiteSettings) {
         this.settings = settings;
-        this.editor.updateOptions({
-            contextmenu: settings.useCustomContextMenu,
-            minimap: {
-                enabled: settings.showMinimap,
-            },
-            fontFamily: settings.editorsFFont,
-            fontLigatures: settings.editorsFLigatures,
-        });
     }
 
-    getCurrentState() {
-        const state = {
+    getCurrentState(): PaneState {
+        return {
             id: this.compilerInfo.compilerId,
-            editorId: this.compilerInfo.editorId,
-            selection: this.selection,
+            compilerName: this.compilerInfo.compilerName,
+            editorid: this.compilerInfo.editorId,
         };
-        this.fontScale.addState(state);
-        return state;
     }
 
     updateState() {
         this.container.setState(this.getCurrentState());
     }
 
+    abstract resize(): void;
+}
+
+/**
+ * Basic container for a tool pane with a monaco editor in Compiler Explorer.
+ *
+ * Type parameter E indicates which monaco editor kind this pane hosts. Common
+ * values are monaco.editor.IDiffEditor and monaco.ICodeEditor
+ *
+ * Type parameter S refers to a state interface for the pane
+ */
+export abstract class MonacoPane<E extends monaco.editor.IEditor, S> extends Pane<S> {
+    editor: E;
+    selection: monaco.Selection;
+    fontScale: FontScale;
+
+    protected constructor(hub: any /* Hub */, container: Container, state: S & MonacoPaneState) {
+        super(hub, container, state);
+
+        const editorRoot = this.domRoot.find('.monaco-placeholder')[0];
+        this.editor = this.createEditor(editorRoot);
+        this.fontScale = new FontScale(this.domRoot, state, this.editor);
+        this.selection = state.selection;
+    }
+
+
+    getCurrentState(): MonacoPaneState {
+        const state = super.getCurrentState() as MonacoPaneState;
+        state.selection = this.selection;
+        this.fontScale.addState(state);
+        return state;
+    }
```

Comment:
I don't really like this type assertion because it's always going to be wrong. I think a better solution is
```suggestion
    getCurrentState(): MonacoPaneState {
        const parent = super.getCurrentState();
        const state: MonacoPaneState = {
            selection: this.selection,
            ...parent,
        };
        this.fontScale.addState(state);
        return state;
    }
```
(assuming this would compile, just did github suggestion)

Reply:
That's what I had to start with I think -- I'll check though!

Reply:
That works! thanks!